### PR TITLE
added type definitions for is-string 1.0

### PR DESCRIPTION
**Pull request by aaronleopold**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by LRotenberger — APPROVED** [review]

File: types/is-string/is-string-tests.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+isString(null); // $Expects false
+isString(false); // $Expects false
+isString(true); // $Expects false
+isString([]); // $Expects false
+isString({}); // $Expects false
+isString(/a/g); // $Expects false
+isString(new Date()); // $Expects false
+isString(42); // $Expects false
+isString(NaN); // $Expects false
+isString(Infinity); // $Expects false
```
Comment (peterblazejewicz) [inline types/is-string/is-string-tests.ts:16]:
```suggestion
isString(''); // $ExpectType boolean
isString(Object('foo')); // $ExpectType boolean

isString(undefined); // $ExpectType boolean
isString(null); // $ExpectType boolean
isString(false); // $ExpectType boolean
isString(true); // $ExpectType boolean
isString([]); // $ExpectType boolean
isString({}); // $ExpectType boolean
isString(/a/g); // $ExpectType boolean
isString(new Date()); // $ExpectType boolean
isString(42); // $ExpectType boolean
isString(NaN); // $ExpectType boolean
isString(Infinity); // $ExpectType boolean
```

Comment (peterblazejewicz) [reply types/is-string/is-string-tests.ts:16]: here we're testing outcomes of type definition, not actual outcome (there is no runtime code execution on DT, only compiler and linter)

File: types/is-string/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for is-string 1.0
+// Project: https://github.com/inspect-js/is-string
+// Definitions by: aaronleopold <https://github.com/aaronleopold>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function isString(value: any): boolean;
```
Comment (peterblazejewicz) [inline types/is-string/index.d.ts:6]:
```suggestion
declare function isString(value: any): value is string;
```
you can mark this function as type predicate:
https://dev.to/daveturissini/aha-understanding-typescript-s-type-predicates-40ha#:~:text=Type%20predicates%20are%20a%20special%20return%20type%20that,Type%20predicates%20are%20expressed%20as%20argumentName%20is%20Type.
It won't affect outcome, but will signal better what's the nature of the function here

Comment (aaronleopold) [reply types/is-string/index.d.ts:6]: Oh thank you for linking the article, I've never seen that before!

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@aaronleopold thx!

**Review by LRotenberger — APPROVED** [review]

**Comment by aaronleopold** [comment]
Ready to merge 🎊

**Pull request merged**